src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -528,10 +528,10 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 		if (!untitledInput.resource || typeof untitledInput.filePath === 'string' || (untitledInput.resource instanceof URI && untitledInput.resource.scheme === Schemas.untitled)) {
 			// {{SQL CARBON EDIT}}
 
-			let mode: string = getFileMode( this.instantiationService, untitledInput.resource);
+			let modeId: string = untitledInput.language ? untitledInput.language : getFileMode( this.instantiationService, untitledInput.resource);
```

Comment:
Just to confirm:
- We wrote this code previously to handle notebook detection?
- The language check handles previously serialized files, since the correct language is saved there?
- This doesn't impact SQL queries in any way?